adapter/repository/in-memory-simple/src/main/java/org/mobilitydata/gtfsvalidator/db/InMemoryGtfsDataRepository.java

```diff
@@ -60,4 +61,23 @@ public Route addRoute(final Route newRoute) throws SQLIntegrityConstraintViolati
             return newRoute;
         }
     }
+
+    private final Map<String, FareAttribute> fareAttributeCollection = new HashMap<>();
+
+    @Override
+    public FareAttribute getFareAttributeByFareId(final String fareId) {
+        return fareAttributeCollection.get(fareId);
+    }
+
+    @Override
+    public FareAttribute addFareAttribute(final FareAttribute newFareAttribute)
+            throws SQLIntegrityConstraintViolationException {
+        if (fareAttributeCollection.containsKey(newFareAttribute.getFareId())) {
+            throw new SQLIntegrityConstraintViolationException("fare attribute must be unique in dataset");
+        } else {
+            String fareId = newFareAttribute.getFareId();
```

Comment:
You don't need to allocate another `String` for `fareId` here - you can eliminate the first line and the next can just be:
`fareAttributeCollection.put(newFareAttribute.getFareId(), newFareAttribute);`